Gauge/Source/Time.hsc

```diff
@@ -71,6 +76,58 @@ withMetrics f = allocaBytes (sizeTimeRecord * 2) $ \ptr -> do
     getRecordPtr ptr2
     (,,) <$> pure a <*> peek ptr <*> peek ptr2
 
+#ifdef __GHCJS__
+data CTimespec = MkCTimespec CTime CLong
+
+instance Storable CTimespec where
+  sizeOf _ = 8
+  alignment _ = 4
```

Comment:
both sizeOf and alignment strikes me as probably working in GHCJS but wrong otherwise. alignment should be 8 and sizeOf = sizeOf CTime + sizeOf CLong

Reply:
I copied this from GHCJS. I am not sure why they are hardcoded. I think `sizeOf = sizeOf CTime + sizeOf CLong` should be fine. I guess the right alignment would be `max (alignment  CTime) (alignment CLong)`. CTime and CLong are perhaps both 4 bytes on GHCJS and therefore alignment of 4 works. `CTimespec` is being read as a 64-bit value in the javascript code handling this.

Reply:
Even `sizeOf CTime + sizeOf CLong` may not be correct in general as there may be padding added due to alignment. So we should also take into account the alignment of each field.